src/accelerate/utils/modeling.py

```diff
@@ -616,7 +616,73 @@ def check_tied_parameters_on_same_device(tied_params, device_map):
             )
 
 
-def find_tied_parameters(model: nn.Module, **kwargs):
+def _get_named_modules(
+    module: torch.nn.Module,
+    memo: Optional[Set[torch.nn.Module]] = None,
+    prefix: str = "",
+    remove_duplicate: bool = True,
+):
+    """
+    Return an iterator over all modules in the network, yielding both the name of the module as well as the module
+    itself. Copied from PyTorch `torch.nn.Module.named_modules` for compatability with torch < 2.0 versions with
+    `remove_duplicate` option added.
+
+    Args:
+        memo: a memo to store the set of modules already added to the result
+        prefix: a prefix that will be added to the name of the module
+        remove_duplicate: whether to remove the duplicated module instances in the result
+            or not
+
+    Yields:
+        (str, Module): Tuple of name and module
+
+    Note:
+        Duplicate modules are returned only once. In the following example, ``l`` will be returned only once.
+
+    Example::
+
+        >>> l = nn.Linear(2, 2) >>> net = nn.Sequential(l, l) >>> for idx, m in enumerate(net.named_modules()): ...
+        print(idx, '->', m)
+
+        0 -> ('', Sequential(
+            (0): Linear(in_features=2, out_features=2, bias=True) (1): Linear(in_features=2, out_features=2, bias=True)
+        )) 1 -> ('0', Linear(in_features=2, out_features=2, bias=True))
```

Comment:
```suggestion
    Example:
    ```python
    >>> from torch import nn
    >>> l = nn.Linear(2, 2)
    >>> net = nn.Sequential(l, l)
    >>> for idx, m in enumerate(net.named_modules()):
    ...   print(idx, '->', m)
    0 -> ('', Sequential(
            (0): Linear(in_features=2, out_features=2, bias=True) (1): Linear(in_features=2, out_features=2, bias=True)
        )) 
    1 -> ('0', Linear(in_features=2, out_features=2, bias=True))
```